core/audits/clickjacking-mitigation.js

```diff
@@ -0,0 +1,159 @@
+  columnDirective: 'Directive',
+  /** Label for a column in a data table; entries will be the severity of an issue with the XFO or CSP header. "XFO" stands for "X-Frame-Options". "CSP" stands for "Content-Security-Policy". */
+  columnSeverity: 'Severity',
+};
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+class ClickjackingMitigation extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'clickjacking-mitigation',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['devtoolsLogs', 'URL'],
+      supportedModes: ['navigation'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<{cspHeaders: string[], xfoHeaders: string[]}>}
+   */
+  static async getRawCspsAndXfo(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const mainResource =
+        await MainResource.request({devtoolsLog, URL: artifacts.URL}, context);
+
+    const cspHeaders = mainResource.responseHeaders
+        .filter(h => {
+          return h.name.toLowerCase() === 'content-security-policy';
+        })
+        .flatMap(h => h.value.split(','))
+        .filter(rawCsp => rawCsp.replace(/\s/g, ''));
+    let xfoHeaders = mainResource.responseHeaders
+                         .filter(h => {
+                           return h.name.toLowerCase() === 'x-frame-options';
+                         })
+                         .flatMap(h => h.value);
+
+    // Sanitize the XFO header value.
+    xfoHeaders = xfoHeaders.map(v => v.toLowerCase().replace(/\s/g, ''));
+
+    return {cspHeaders, xfoHeaders};
+  }
+
+  /**
+   * @param {string | undefined} directive
+   * @param {LH.IcuMessage | string} findingDescription
+   * @param {LH.IcuMessage=} severity
+   * @return {LH.Audit.Details.TableItem}
+   */
+  static findingToTableItem(directive, findingDescription, severity) {
+    return {
+      directive: directive,
+      description: findingDescription,
+      severity,
+    };
+  }
+
+  /**
+   * @param {string[]} cspHeaders
+   * @param {string[]} xfoHeaders
+   * @return {{score: number, results: LH.Audit.Details.TableItem[]}}
+   */
+  static constructResults(cspHeaders, xfoHeaders) {
+    const rawXfo = [...xfoHeaders];
+    const allowedDirectives = ['deny', 'sameorigin'];
+
+    // If there is none of the two headers, return early.
+    if (!rawXfo.length && !cspHeaders.length) {
+      return {
+        score: 0,
+        results: [{
+          severity: str_(i18n.UIStrings.itemSeverityHigh),
+          description: str_(UIStrings.noClickjackingMitigation),
+          directive: undefined,
+        }],
+      };
+    }
```

Comment:
```suggestion
```

This is redundant, if the header lists are empty we will just skip to the bottom anyway.